lib/Mongo/MongoGridFSFile.php

```diff
@@ -71,5 +109,34 @@ public function getBytes() {}
      * @link http://php.net/manual/en/mongogridfsfile.getresource.php
      * @return resource Returns a resource that can be used to read the file with
      */
-    public function getResource() {}
+    public function getResource()
+    {
+        $handle = tmpfile();
+        $this->writeFromRessource($handle);
+        fseek($handle, 0);
+        return $handle;
+    }
+
+    private function getChunks()
+    {
+        return $chunks = $this->gridfs->chunks->find(
+            ['file_id' => new \MongoDB\BSON\ObjectID((string) $this->file['_id'])],
```

Comment:
@alcaeus I don't think I should have to convert type here, I think we got some new types leaking somewhere.